h2o-algos/src/test/java/hex/rulefit/RuleFitTest.java

```diff
@@ -480,6 +486,55 @@ public void testCarsRules() {
         }
     }
 
+    @Test
+    public void testCarsRulesValidation() {
+        try {
+            Scope.enter();
+            final Frame fr = Scope.track(parseTestFile("smalldata/testng/cars_train.csv"));
+            final Frame valid = Scope.track(parseTestFile("smalldata/testng/cars_test.csv"));
+            RuleFitModel.RuleFitParameters params = new RuleFitModel.RuleFitParameters();
+            params._seed = 1234;
+            params._response_column = "power (hp)";
+            params._ignored_columns = new String[]{"name"};
+            params._train = fr._key;
+            params._valid = valid._key;
+            params._max_num_rules = 200;
+            params._max_rule_length = 5;
+            params._model_type = RuleFitModel.ModelType.RULES;
+            params._min_rule_length = 1;
+
+            RuleFitModel model = new RuleFit( params).trainModel().get();
+            Scope.track_generic(model);
+
+            System.out.println("Intercept: \n" + model._output._intercept[0]);
+            System.out.println(model._output._rule_importance);
+
+            final Frame fr2 = Scope.track(model.score(fr));
+
+            GLMModel.GLMParameters glmParameters = model.glmModel._parms;
+            glmParameters._train = fr._key;
+            glmParameters._valid = valid._key;
+            glmParameters._response_column = model._parms._response_column;
+            final GLMModel glmModel = new GLM(glmParameters).trainModel().get();
+            Scope.track_generic(glmModel);
+            Scope.track(glmModel.score(fr));
+
+            Assert.assertTrue(model.testJavaScoring(fr,fr2,1e-4));
+
+            ScoringInfo RuleFitScoringInfo = model.glmModel.getScoringInfo()[0];
```

Comment:
is the correctness of validation metrics tested anywhere? as a bare minimum you can set training and validation to be the same frame and check that the metrics are identical

Reply:
60efcf1